logback/src/main/java/com/grafana/loki/HttpHandler.java

```diff
@@ -0,0 +1,60 @@
+package com.grafana.loki;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpRequest.Builder;
+import java.net.http.HttpResponse;
+
+public class HttpHandler {
+	private static HttpClient client;
+
+	public static void sendLoki(String tenantID, URI url, byte[] body) {
+		int retryCount = 3;
+		int retryDelay = 2000;
+		if (client == null) {
+			client = HttpClient.newHttpClient();
+		}
+
+		Builder builder = HttpRequest.newBuilder().uri(URI.create(url.toString()))
+				.POST(HttpRequest.BodyPublishers.ofByteArray(body));
+		builder.header("Content-Type", "application/x-protobuf");
+
+		if (tenantID != "") {
+			builder.header("X-Scope-OrgID", tenantID);
+		}
+		HttpRequest request = builder.build();
+
+		RetryOnException retryHandler = new RetryOnException(retryCount, retryDelay);
+		while (true) {
+			HttpResponse<String> response;
+			try {
+				response = doRequest(request);
+				if (response.statusCode() > 0 && response.statusCode() != 429 && response.statusCode() / 100 != 5) {
+					break;
+				}
```

Comment:
I think with the above change on the while loop, we would need an `else` statement here with a call to `retryHandler.exceptionOcurred()`

Reply:
Will make the change